### PR TITLE
refactor: removing support for JUnit4

### PR DESCRIPTION
**Pull request by elbiocaetano**

### CODE REVIEW

File: sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 public class AuthClassExtension extends AbstractAuth
     implements BeforeAllCallback, AfterAllCallback {
 
-  private static final String AUTH_RULE_ENV_KEY = "AUTH_RULE";
+  public static final String AUTH_RULE_ENV_KEY = "AUTH_RULE";
```
Comment (eikenilsknopp) [inline sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java:17]: While we're at it: Is this naming still appropriate (rule -> extension)?

File: sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java [hunk 2]
```diff
@@ -36,9 +36,9 @@
  * A bundle used to generate and serve API-first or code-first OpenAPI 3 files at the {@code
  * openapi.yaml} or {@code openapi.json} HTTP resources.
  *
- * <h3>Example Usage</h3>
+ * <p>Example Usage:
```
Comment (eikenilsknopp) [inline sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java:39]: Shouldn’t the paragraph be closed with </p>?

File: sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java:17]: I think it's a good opportunity to rename the variable name + value. Consequences for customers: Double check if you use that variable in any one of your `test-config.yaml`s and rename it accordingly.

File: sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java (hunk 2 above)
Comment (christopher-cudennec) [reply sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java:39]: Javadoc is not HTML; a single `<p>` is correct according to [SO](https://stackoverflow.com/questions/33787753/javadoc-closing-tags) / the official documentation.

**Review by christopher-cudennec — COMMENTED** [review]
3000+ lines of code removed 🥳

**Pull request merged**